src/ripple/rpc/handlers/AccountInfo.cpp

```diff
@@ -125,6 +125,15 @@ doAccountInfo(RPC::JsonContext& context)
         }
         result[jss::account_flags] = std::move(acctFlags);
 
+        // The document states that signer_lists is a bool, however
+        // assigning any string value works. Do not allow this
+        // This for api v2 onwards
+        if (!params[jss::signer_lists].isBool() && context.apiVersion > 1)
```

Comment:
do we need to check the `context.apiVersion` here? We assume that all the validators are running the latest version of the rippled codebase. Otherwise, the laggards will be incompatible with the rest of the network right?

Reply:
This is a situation where we don't need to worry about validators very much, at least on the main net.  This is an RPC query.  It's not typical for main net validators to be required also handle very many RPC queries.

The `context.api` can be a value passed in by the caller as shown in the test where it says `"\"api_version\": 2`.  So, yes, we should check the `api_version`.  It can be different from one RPC call to the next.

Reply:
i see okay, thanks for the clarification 👍 
I'm guessing the version of rippled software is preserved in a different variable right?

Reply:
Yes, the software version (e.g., `1.12.0-b1`) is elsewhere.  The code never needs to check the software version because it is compiled into the binary with whatever software version it has.  The `api_version` is a different kind of beast.

Reply:
i see, okay